test/orange/g4org/SolidConverter.test.cc

```diff
@@ -734,47 +734,76 @@ TEST_F(SolidConverterTest, polycone)
-            {0, -1.01, 1.1},
+            {-0.1, 0, 26.},
+            {-0.1, 0, -11.},
+            {10, 0, -5},
+            {10, 10, -5},
+            {-3.5, 0, 23},
+            {-4.5, 0, 23},
+            {-5.5, 0, 23},
         });
-    // Rotate 60 degrees
+
+    // One-sided shape, with interior
     this->build_and_test(
         G4Polyhedra(
-            "tri", 60 * deg, 360 * deg, 3, std::size(z2), z2, rmin2, rmax2),
-        R"json({"_type":"transformed","daughter":{"_type":"shape","interior":{"_type":"prism","apothem":0.9999999999999999,"halfheight":2.0,"num_sides":3,"orientation":0.5},"label":"tri"},"transform":{"_type":"translation","data":[0.0,0.0,3.0]}})json");
-    // Rotate 90 degrees
+            "polyhedra", -10 * deg, 20 * deg, 1, std::size(z), z, rmin, rmax),
+        R"json({"_type":"all","daughters":[{"_type":"stackedextrudedpolygon","polygon":[[3.0,-0.5289809421253949],[3.0,0.5289809421253949],[0.0,0.0]],"polyline":[[0.0,0.0,-1.0],[0.0,0.0,0.0],[0.0,0.0,2.0],[0.0,0.0,2.5]],"scaling":[1.0,0.33333333333333337,0.6666666666666667,0.16666666666666669]},{"_type":"negated","daughter":{"_type":"stackedextrudedpolygon","polygon":[[3.0,-0.5289809421253949],[3.0,0.5289809421253949],[0.0,0.0]],"polyline":[[0.0,0.0,-1.0],[0.0,0.0,0.0],[0.0,0.0,2.0],[0.0,0.0,2.5]],"scaling":[0.6666666666666667,0.13333333333333336,0.5666666666666668,0.13333333333333336]},"label":""}],"label":"polyhedra"})json",
+        {
+            {-19, 0, -1},
+            {-25, 0, -1},
+            {-31, 0, -1},
+            {-20, 10, -1},
+        });
+
+    // Two-sided shape, with interior
     this->build_and_test(
         G4Polyhedra(
-            "tri", 90 * deg, 360 * deg, 3, std::size(z2), z2, rmin2, rmax2),
-        R"json({"_type":"transformed","daughter":{"_type":"shape","interior":{"_type":"prism","apothem":0.9999999999999999,"halfheight":2.0,"num_sides":3,"orientation":0.75},"label":"tri"},"transform":{"_type":"translation","data":[0.0,0.0,3.0]}})json");
+            "polyhedra", 0 * deg, 180 * deg, 2, std::size(z), z, rmin, rmax),
+        R"json({"_type":"all","daughters":[{"_type":"stackedextrudedpolygon","polygon":[[4.242640687119285,0.0],[2.5978681687064796e-16,4.242640687119285],[-4.242640687119285,5.195736337412959e-16],[0.0,0.0]],"polyline":[[0.0,0.0,-1.0],[0.0,0.0,0.0],[0.0,0.0,2.0],[0.0,0.0,2.5]],"scaling":[1.0,0.3333333333333333,0.6666666666666666,0.16666666666666666]},{"_type":"negated","daughter":{"_type":"stackedextrudedpolygon","polygon":[[4.242640687119285,0.0],[2.5978681687064796e-16,4.242640687119285],[-4.242640687119285,5.195736337412959e-16],[0.0,0.0]],"polyline":[[0.0,0.0,-1.0],[0.0,0.0,0.0],[0.0,0.0,2.0],[0.0,0.0,2.5]],"scaling":[0.6666666666666666,0.13333333333333336,0.5666666666666668,0.13333333333333336]},"label":""}],"label":"polyhedra"})json",
+        {
+            {19, 1, -1},
+            {25, 1, -1},
+            {31, 1, -1},
+            {0, 31, -1},
+            {2, 29, -1},
+        });
```

Comment:
Can you also please add the actual CMS shapes described in https://github.com/celeritas-project/celeritas/issues/2046 ? That'll help verification even if you don't put manual points (since the test also does random sampling, 4k points by default).

Reply:
Alas, the z-planes for `HEC10x7f1fffce6500` are not strictly monotonically increasing: there are repeated z-planes! We don't support with `StackedExtrudedSolid` because it would require making cones with zero height.

For now the terrible hack of simply bumped these planes by 1e-12, results in passing test with 4k points checked.

I think I can add support for this with `StackedExtrudedPolygon` but I will have to look.

Reply:
Ughh, the other CMS one is even worse. We hit:
```
167: C++ exception with description "/Users/veb/opt/celeritas/celeritas/src/orange/orangeinp/detail/CsgUnitBuilder.cc:91:
167: celeritas: internal assertion failed: static_cast<bool>(bzone.interior) == static_cast<bool>(existing.bounds.interior)" thrown in the test body.
```
This will take some tinkering.